fobi_custom/plugins/form_elements/fields/observational/widgets.py

```diff
@@ -0,0 +1,36 @@
+import re
+from django import forms
+
+
+class ObservationalWidget(forms.MultiWidget):
+    # template_name = 'observational_widget.html'
+
+    def __init__(self, choices, *args, **kwargs):
+        self.choices = choices
+        self.widgets = []
+        for _, choice in choices:
+            self.widgets = self.widgets + [forms.NumberInput()]
+
+        super(ObservationalWidget, self).__init__(self.widgets, *args, **kwargs)
+
+    def render(self, name, value, attrs=None, renderer=None):
+        context = self.get_context(name, value, attrs)
+        rendered_html = self._render(self.template_name, context, renderer)
+
+        split_html = re.findall('\<input.*?\>', rendered_html)
+        processed_html = ['<label>' + self.get_choice(self.choices, input, split_html) + '</label>' + input for input in split_html]
```

Comment:
I can see why this was making you uneasy. Let's have a quick call to talk about alternatives.

Reply:
The custom form renderers have me stalled, and I'm not sure this is worth digging into more: https://docs.djangoproject.com/en/2.2/ref/forms/renderers/

@jeancochrane could you take a look?

I also realized I was wrong about collected data containing `<input` causing a problem—we'd only run into issues if one of our subwidget labels included the text `<input`. Those aren't currently exposed to user input (they're the `choices` [described in the PLDP](https://github.com/datamade/django-pldp/blob/master/pldp/forms.py)), but the protocol does recommend allowing users to add their own options for some form elements. If we keep this method of producing labels for subwidgets, I think that should be a separate issue.

Reply:
Thanks for taking a closer look at this. I'd be interested to dig deeper into it but I don't have high confidence that I'll find a solution that you haven't, and I don't want to hold up this PR.

I say let's merge this and update the widget rendering if I can find a better path (or if we run itno errors down the line). Does that sound OK to you @beamalsky?

Reply:
Sounds good!

Reply:
Opened an issue for this: #116